src/rust/engine/src/nodes.rs

```diff
@@ -916,8 +939,13 @@ impl WrappedNode for Task {
     }
 
     if result_type == product {
-      let maybe_new_level = Self::compute_new_workunit_level(can_modify_workunit, &result_val);
-      Ok((result_val, maybe_new_level))
+      let new_level = Self::compute_new_workunit_level(can_modify_workunit, &result_val);
+      let info_msg = Self::compute_info_msg(can_modify_workunit, &result_val);
+      Ok(PythonRuleOutput {
+        value: result_val,
+        new_level,
+        info_msg,
+      })
```

Comment:
It might be a little more-efficient/less-redundant for these two methods to be merged together into one method that consumes the value and returns a `PythonRuleOutput`?